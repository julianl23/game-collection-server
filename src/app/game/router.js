import { getList, put } from './controller';

const routes = [
  {
    path: '/api/games',
    method: 'GET',
    options: {
      log: {
        collect: true
      },
      auth: false
    },
    ...getList
  },
  {
    path: '/api/games',
    method: 'PUT',
    options: {
      auth: false
    },
    ...put
  }
];

export default routes;
