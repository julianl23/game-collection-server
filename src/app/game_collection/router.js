import { get, addGame } from './controller';

const routes = [
  {
    path: '/api/collection',
    method: 'GET',
    options: {
      log: {
        collect: true
      },
      auth: false
    },
    ...get
  },
  {
    path: '/api/collection/add',
    method: 'PUT',
    options: {
      log: {
        collect: true
      }
    },
    ...addGame
  }
];

export default routes;
