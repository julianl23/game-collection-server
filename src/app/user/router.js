import { getList, put, login } from './controller';

const routes = [
  {
    path: '/api/users',
    method: 'GET',
    options: {
      log: {
        collect: true
      },
      auth: false
    },
    ...getList
  },
  {
    path: '/api/users',
    method: 'PUT',
    options: {
      log: {
        collect: true
      },
      cors: {
        origin: ['*'],
        credentials: true
      },
      auth: false
    },
    ...put
  },
  {
    path: '/api/users/login',
    method: 'POST',
    options: {
      log: { collect: true },
      cors: {
        origin: ['*'],
        credentials: true
      },
      auth: false
    },
    ...login
  }
];

export default routes;
