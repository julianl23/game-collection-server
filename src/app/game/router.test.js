import router from './router';

jest.mock('./controller');

describe('game', () => {
  describe('router', () => {
    describe('/api/games', () => {
      it('has the correct structure', () => {
        expect(router).toMatchSnapshot();
      });
    });
  });
});
