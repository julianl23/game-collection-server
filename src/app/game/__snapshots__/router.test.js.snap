// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game router /api/games has the correct structure 1`] = `
Array [
  Object {
    "handler": [MockFunction],
    "method": "GET",
    "options": Object {
      "auth": false,
      "log": Object {
        "collect": true,
      },
    },
    "path": "/api/games",
  },
  Object {
    "handler": [MockFunction],
    "method": "PUT",
    "options": Object {
      "auth": false,
    },
    "path": "/api/games",
  },
  Object {
    "handler": [MockFunction],
    "method": "GET",
    "options": Object {
      "auth": false,
      "log": Object {
        "collect": false,
      },
    },
    "path": "/api/games/igdbTest",
  },
]
`;
